src/dig.c

```diff
@@ -85,6 +85,11 @@ mkcavepos(x, y, dist, waslit, rockit)
     if(Blind)
 	feel_location(x, y);
     else newsym(x,y);
+
+	if (rockit)
+		block_point(x, y);
+	else
+		unblock_point(x, y);
```

Comment:
I think this needs to check if(!does_block()). There might hypothetically be a boulder on that spot.